adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/X509AuthHandler.java

```diff
@@ -0,0 +1,233 @@
+/**
```

Comment:
This class copies a lot from the http/X509AuthHandler. Maybe with some refactoring we can reuse the existing code instead of duplicating it?

Reply:
Yes, I thought so too. However, the only way I can imagine reusing the code would be to factor it out into some static helper methods in a utility class. Or do you have another idea?

Reply:
Yeah, a small utility class with parseCredentials(Certificate[] path) and other duplicated methods would be a good approach.

Reply:
Ok, I will try that (tomorrow) ...